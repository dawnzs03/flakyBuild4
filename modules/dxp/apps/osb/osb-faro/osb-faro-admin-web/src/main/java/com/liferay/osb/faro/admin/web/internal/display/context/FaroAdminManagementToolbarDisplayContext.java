/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.admin.web.internal.display.context;

import com.liferay.frontend.taglib.clay.servlet.taglib.display.context.SearchContainerManagementToolbarDisplayContext;
import com.liferay.frontend.taglib.clay.servlet.taglib.util.CreationMenu;
import com.liferay.petra.string.StringPool;
import com.liferay.portal.kernel.dao.search.SearchContainer;
import com.liferay.portal.kernel.language.LanguageUtil;
import com.liferay.portal.kernel.portlet.LiferayPortletRequest;
import com.liferay.portal.kernel.portlet.LiferayPortletResponse;
import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
import com.liferay.portal.kernel.security.permission.PermissionChecker;
import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
import com.liferay.portal.kernel.theme.ThemeDisplay;
import com.liferay.portal.kernel.util.ParamUtil;
import com.liferay.portal.kernel.util.WebKeys;

import javax.portlet.ActionRequest;
import javax.portlet.PortletURL;
import javax.portlet.RenderResponse;

import javax.servlet.http.HttpServletRequest;

/**
 * @author Shinn Lok
 */
public class FaroAdminManagementToolbarDisplayContext
	extends SearchContainerManagementToolbarDisplayContext {

	public FaroAdminManagementToolbarDisplayContext(
		HttpServletRequest httpServletRequest,
		LiferayPortletRequest liferayPortletRequest,
		LiferayPortletResponse liferayPortletResponse,
		RenderResponse renderResponse, SearchContainer<?> searchContainer) {

		super(
			liferayPortletRequest, liferayPortletResponse, httpServletRequest,
			searchContainer);

		_renderResponse = renderResponse;

		_themeDisplay = (ThemeDisplay)httpServletRequest.getAttribute(
			WebKeys.THEME_DISPLAY);
	}

	@Override
	public String getClearResultsURL() {
		return PortletURLBuilder.create(
			getPortletURL()
		).setKeywords(
			StringPool.BLANK
		).buildString();
	}

	public CreationMenu getCreationMenu() {
		PermissionChecker permissionChecker =
			PermissionThreadLocal.getPermissionChecker();

		if (!permissionChecker.isOmniadmin()) {
			return null;
		}

		PortletURL portletURL = PortletURLBuilder.create(
			_renderResponse.createActionURL()
		).setRedirect(
			ParamUtil.getString(
				liferayPortletRequest.getHttpServletRequest(), "redirect",
				_themeDisplay.getURLCurrent())
		).buildPortletURL();

		return new CreationMenu() {
			{
				addDropdownItem(
					dropdownItem -> {
						dropdownItem.setHref(
							portletURL, ActionRequest.ACTION_NAME,
							"/faro_admin/refresh_project", "groupId", -1);
						dropdownItem.setLabel(
							LanguageUtil.get(request, "refresh-all-projects"));
					});
			}
		};
	}

	@Override
	public String getInfoPanelId() {
		return "infoPanelId";
	}

	@Override
	public String getSearchActionURL() {
		return PortletURLBuilder.create(
			getPortletURL()
		).setParameter(
			"orderByCol", getOrderByCol()
		).setParameter(
			"orderByType", getOrderByType()
		).buildString();
	}

	@Override
	public String getSearchContainerId() {
		return "faro_admin";
	}

	@Override
	public Boolean isDisabled() {
		return false;
	}

	@Override
	protected String[] getNavigationKeys() {
		return new String[] {
			"all", "basic", "business", "enterprise", "inactive", "offline",
			"usage-limit-approaching", "usage-limit-exceeded"
		};
	}

	@Override
	protected String[] getOrderByKeys() {
		return new String[] {
			"createDate", "individualsUsage", "name", "pageViewsUsage"
		};
	}

	private final RenderResponse _renderResponse;
	private final ThemeDisplay _themeDisplay;

}