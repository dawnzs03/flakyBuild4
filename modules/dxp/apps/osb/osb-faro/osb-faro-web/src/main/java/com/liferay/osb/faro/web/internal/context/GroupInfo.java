/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.context;

/**
 * @author Marcellus Tavares
 */
public class GroupInfo {

	public GroupInfo(long groupId) {
		_groupId = groupId;
	}

	public long getGroupId() {
		return _groupId;
	}

	private final long _groupId;

}