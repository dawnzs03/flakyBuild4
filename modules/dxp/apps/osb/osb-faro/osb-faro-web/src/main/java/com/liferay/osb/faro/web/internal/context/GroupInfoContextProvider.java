/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.context;

import com.github.scribejava.core.exceptions.OAuthException;

import com.liferay.expando.kernel.model.ExpandoBridge;
import com.liferay.oauth2.provider.model.OAuth2Authorization;
import com.liferay.oauth2.provider.service.OAuth2AuthorizationLocalService;

import java.util.Date;

import javax.servlet.http.HttpServletRequest;

import javax.ws.rs.ext.Provider;

import org.apache.cxf.jaxrs.ext.ContextProvider;
import org.apache.cxf.message.Message;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marcellus Tavares
 */
@Component(service = GroupInfoContextProvider.class)
@Provider
public class GroupInfoContextProvider implements ContextProvider<GroupInfo> {

	@Override
	public GroupInfo createContext(Message message) {
		HttpServletRequest httpServletRequest =
			(HttpServletRequest)message.getContextualProperty("HTTP.REQUEST");

		String authorization = httpServletRequest.getHeader("Authorization");

		if (authorization == null) {
			throw new IllegalStateException(
				"Authorization Header is not available");
		}

		try {
			OAuth2Authorization oAuth2Authorization =
				_oAuth2AuthorizationLocalService.
					getOAuth2AuthorizationByAccessTokenContent(
						authorization.substring(7));

			Date currentDate = new Date(System.currentTimeMillis());

			if (currentDate.after(
					oAuth2Authorization.getAccessTokenExpirationDate())) {

				throw new OAuthException(
					"Your access token is invalid. Please check your token " +
						"and try again.");
			}

			ExpandoBridge expandoBridge =
				oAuth2Authorization.getExpandoBridge();

			return new GroupInfo(
				(long)expandoBridge.getAttribute("groupId", false));
		}
		catch (Exception exception) {
			throw new IllegalStateException(
				"Unable to fetch the OAuth2Authorization with access token " +
					authorization.substring(7),
				exception);
		}
	}

	@Reference
	private OAuth2AuthorizationLocalService _oAuth2AuthorizationLocalService;

}