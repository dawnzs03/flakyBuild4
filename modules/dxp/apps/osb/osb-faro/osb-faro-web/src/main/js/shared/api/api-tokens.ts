import sendRequest from 'shared/util/request';
import {RESTParams} from 'shared/types';

export function generate({
	expiresIn,
	groupId
}: RESTParams & {expiresIn: string}) {
	return sendRequest({
		method: 'POST',
		path: `main/${groupId}/oauth2/tokens/new?expiresIn=${expiresIn}`
	});
}

export function search({groupId}: RESTParams) {
	return sendRequest({
		method: 'GET',
		path: `main/${groupId}/oauth2/tokens`
	});
}

export function revoke({groupId, token}: RESTParams & {token: string}) {
	return sendRequest({
		method: 'POST',
		path: `main/${groupId}/oauth2/tokens/${token}/revoke`
	});
}
