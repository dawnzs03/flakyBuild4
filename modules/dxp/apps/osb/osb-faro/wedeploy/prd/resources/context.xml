<Context>
	<Valve className="org.apache.catalina.valves.rewrite.RewriteValve" />
	<WatchedResource>${catalina.base}/conf/web.xml</WatchedResource>
	<WatchedResource>WEB-INF/tomcat-web.xml</WatchedResource>
	<WatchedResource>WEB-INF/web.xml</WatchedResource>
</Context>