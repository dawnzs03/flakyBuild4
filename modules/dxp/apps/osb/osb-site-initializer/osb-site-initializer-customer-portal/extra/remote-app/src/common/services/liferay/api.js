/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {Liferay} from '.';

const HEADLESS_DELIVERY_BASE_URL_ = `${window.location.origin}/o/headless-delivery/v1.0`;
const HEADLESS_BASE_URL = `${window.location.origin}/o/`;

const fetchHeadless = async ({resolveAsJson = true, url}) => {
	// eslint-disable-next-line @liferay/portal/no-global-fetch
	const response = await fetch(`${HEADLESS_DELIVERY_BASE_URL_}${url}`, {
		headers: {
			'Accept-Language': Liferay.ThemeDisplay.getBCP47LanguageId(),
			'Cache-Control': 'max-age=30, stale-while-revalidate=30',
			'x-csrf-token': Liferay.authToken,
		},
	});

	if (resolveAsJson) {
		return response.json();
	}

	return response;
};

const getHighPriorityContacts = async (filter) => {
	// eslint-disable-next-line @liferay/portal/no-global-fetch
	const response = await fetch(
		`${HEADLESS_BASE_URL}${`c/highprioritycontactses/?nestedFields=user&filter=${filter}`}`,
		{
			headers: {
				'Accept-Language': Liferay.ThemeDisplay.getBCP47LanguageId(),
				'Cache-Control': 'max-age=30, stale-while-revalidate=30',
				'x-csrf-token': Liferay.authToken,
			},
		}
	);

	return response.json();
};

export {getHighPriorityContacts, fetchHeadless};
