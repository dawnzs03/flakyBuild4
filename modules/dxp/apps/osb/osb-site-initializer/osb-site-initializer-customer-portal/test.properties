##
## Test Batch
##

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57-jdk8][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]}

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer Customer Portal") OR \
            (testray.main.component.name ~ "OSB Site Initializer Customer Portal")\
        )