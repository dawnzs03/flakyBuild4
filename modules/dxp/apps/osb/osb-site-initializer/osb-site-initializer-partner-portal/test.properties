##
## Salesforce
##

    #site.initializer.partner.portal.login.url=
    #site.initializer.partner.portal.consumer.key=
    #site.initializer.partner.portal.consumer.secret=
    #site.initializer.partner.portal.username=
    #site.initializer.partner.portal.password=

##
## Test Batch
##

    #
    # Portal Hotfix Release
    #

    test.batch.run.property.query[functional-bundle-tomcat-mysql57-jdk8][test-portal-hotfix-release]=\
        ${test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]}

    #
    # Relevant
    #

    test.batch.run.property.query[functional-tomcat*-mysql*-jdk8][relevant]=\
        (portal.acceptance == true) AND \
        (\
            (testray.component.names ~ "OSB Site Initializer Partner Portal") OR \
            (testray.main.component.name ~ "OSB Site Initializer Partner Portal")\
        )

##
## Testray
##

    testray.main.component.name=OSB Site Initializer Partner Portal