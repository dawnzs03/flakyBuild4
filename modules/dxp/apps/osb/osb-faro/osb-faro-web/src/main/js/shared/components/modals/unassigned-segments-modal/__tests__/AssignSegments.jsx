import * as API from 'shared/api';
import AssignSegments from '../AssignSegments';
import React from 'react';
import {ChannelContext} from 'shared/context/channel';
import {
	cleanup,
	fireEvent,
	render,
	waitForDomChange
} from '@testing-library/react';
import {mockChannelContext} from 'test/mock-channel-context';
import {mockSegment} from 'test/data';
import {noop} from 'lodash';
import {StaticRouter} from 'react-router-dom';
import {UnassignedSegmentsContext} from 'shared/context/unassignedSegments';

jest.unmock('react-dom');

const mockedContext = {
	unassignedSegments: [
		mockSegment(1, {channelId: null}),
		mockSegment(2, {channelId: null})
	],
	unassignedSegmentsDispatch: jest.fn(),
	unassignedSegmentsTriggered: false
};

const DefaultComponent = props => (
	<UnassignedSegmentsContext.Provider value={mockedContext}>
		<ChannelContext.Provider value={mockChannelContext()}>
			<StaticRouter>
				<AssignSegments groupId='123' onClose={noop} {...props} />
			</StaticRouter>
		</ChannelContext.Provider>
	</UnassignedSegmentsContext.Provider>
);

describe('AssignSegments', () => {
	afterEach(() => {
		cleanup();
		jest.useRealTimers();
	});

	it('should render', () => {
		jest.useFakeTimers();
		const {container, getByTestId, getByText} = render(
			<DefaultComponent />
		);
		fireEvent.click(getByTestId('select-1'));

		jest.runAllTimers();

		expect(getByText('Unassigned')).toBeTruthy();
		expect(getByText('Delete')).toBeTruthy();
		expect(getByText('Channel 1')).toBeTruthy();
		expect(getByText('Channel 2')).toBeTruthy();

		expect(container).toMatchSnapshot();
	});

	it('should run close from OnClose prop', async () => {
		const spy = jest.fn();

		const {getByTestId, getByText} = render(
			<DefaultComponent onClose={spy} />
		);

		fireEvent.click(getByText('Skip for Now'));

		fireEvent.click(getByTestId('submit-button'));

		expect(spy).toBeCalled();
	});

	it('it should enable done button when a valid value is selected', async () => {
		const {getByTestId, getByText} = render(<DefaultComponent />);

		fireEvent.click(getByTestId('select-1'));

		jest.runAllTimers();

		fireEvent.click(getByText('Channel 1'));

		const button = getByTestId('submit-button');

		expect(button).not.toBeDisabled();
	});

	it('should call api functions with Channel 1 args', async () => {
		const {container, getByTestId, getByText} = render(
			<DefaultComponent />
		);

		fireEvent.click(getByTestId('select-1'));

		jest.runAllTimers();

		fireEvent.click(getByText('Channel 1'));

		fireEvent.click(getByTestId('submit-button'));

		await waitForDomChange(container.querySelector('.assign-segments'));

		expect(API.individualSegment.updateChannel).toBeCalledWith({
			channelId: '1',
			groupId: '123',
			id: '1'
		});
	});

	it('should call api functions with Delete args', async () => {
		const {container, getByTestId, getByText} = render(
			<DefaultComponent />
		);

		fireEvent.click(getByTestId('select-1'));

		jest.runAllTimers();

		fireEvent.click(getByText('Delete'));

		fireEvent.click(getByTestId('submit-button'));

		await waitForDomChange(container.querySelector('.assign-segments'));

		expect(API.individualSegment.delete).toBeCalledWith({
			groupId: '123',
			id: '1'
		});
	});
});
