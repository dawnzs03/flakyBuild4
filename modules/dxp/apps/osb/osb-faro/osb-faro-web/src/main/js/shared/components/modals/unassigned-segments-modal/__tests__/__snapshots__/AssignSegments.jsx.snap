// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignSegments should render 1`] = `
<div>
  <div
    class="assign-segments"
  >
    <div
      class="modal-body d-flex flex-column align-items-center"
    >
      <h2
        aria-hidden="true"
        data-aria-hidden="true"
      >
        Assign Existing Segments
      </h2>
      <span
        aria-hidden="true"
        class="subtitle"
        data-aria-hidden="true"
      >
        Your existing segments will be hidden until they have been assigned. A segment can only belong to a single property.
      </span>
      <div
        class="assign-segments-table-wrapper"
      >
        <div
          class="table-responsive table-root flex-grow-1"
        >
          <table
            class="table table-autofit table-hover table-heading-nowrap table-nowrap"
          >
            <thead
              aria-hidden="true"
              data-aria-hidden="true"
            >
              <tr>
                <th
                  class="table-head-title"
                >
                  <button
                    class="inline-item text-truncate-inline btn btn-unstyled"
                    type="button"
                  >
                    <span
                      class="text-truncate"
                    >
                      Segment Name
                    </span>
                    <span
                      class="inline-item inline-item-after"
                    >
                      <svg
                        class="lexicon-icon lexicon-icon-order-arrow-up icon-root"
                        role="presentation"
                      >
                        <use
                          xlink:href="#order-arrow-up"
                        />
                      </svg>
                    </span>
                  </button>
                </th>
                <th
                  class="table-head-title"
                >
                  <button
                    class="inline-item text-truncate-inline btn btn-unstyled"
                    type="button"
                  >
                    <span
                      class="text-truncate"
                    >
                      Created by
                    </span>
                  </button>
                </th>
                <th
                  class="table-head-title"
                />
              </tr>
            </thead>
            <tbody>
              <tr
                class=""
              >
                <td
                  aria-hidden="true"
                  data-aria-hidden="true"
                >
                  Seattle1
                </td>
                <td
                  aria-hidden="true"
                  data-aria-hidden="true"
                >
                  -
                </td>
                <td>
                  <button
                    aria-activedescendant=""
                    aria-expanded="true"
                    aria-haspopup="listbox"
                    aria-owns="clay-id-2"
                    class="form-control form-control-select form-control-select-secondary show"
                    data-testid="select-1"
                    required=""
                    role="combobox"
                    style=""
                    tabindex="0"
                    type="button"
                  >
                    Select an option
                  </button>
                </td>
              </tr>
              <tr
                aria-hidden="true"
                class=""
                data-aria-hidden="true"
              >
                <td>
                  Seattle2
                </td>
                <td>
                  -
                </td>
                <td>
                  <button
                    aria-activedescendant=""
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="form-control form-control-select form-control-select-secondary"
                    data-testid="select-2"
                    required=""
                    role="combobox"
                    tabindex="0"
                    type="button"
                  >
                    Select an option
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="modal-footer d-flex justify-content-end"
      data-aria-hidden="true"
    >
      <button
        class="btn btn-secondary"
        type="button"
      >
        Skip for Now
      </button>
      <button
        class="btn btn-primary"
        data-testid="submit-button"
        disabled=""
        type="button"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;
